contrib/sarplus/README.md

```diff
@@ -0,0 +1,191 @@
+        'rating':  [1, 1, 1, 1, 1],
+    }))
+   
+# spark dataframe with user/item tuples
+test_df = spark.createDataFrame(
+      pd.DataFrame({
+        'user_id': [1, 3],
+        'item_id': [1, 3],
+        'rating':  [1, 1],
+    }))
+    
+model = SARPlus(spark, col_user='user_id', col_item='item_id', col_rating='rating', col_timestamp='timestamp')
+model.fit(train_df, similarity_type='jaccard')
+
+
+model.recommend_k_items(test_df, 'sarplus_cache', top_k=3).show()
+
+# For databricks
+# model.recommend_k_items(test_df, 'dbfs:/mnt/sarpluscache', top_k=3).show()
+```
+
+## Jupyter Notebook
+
+Insert this cell prior to the code above.
+
+```python
+import os
+
+SUBMIT_ARGS = "--packages microsoft:sarplus:0.2.6 pyspark-shell"
+os.environ["PYSPARK_SUBMIT_ARGS"] = SUBMIT_ARGS
+
+from pyspark.sql import SparkSession
+
+spark = (
+    SparkSession.builder.appName("sample")
+    .master("local[*]")
+    .config("memory", "1G")
+    .config("spark.sql.shuffle.partitions", "1")
+    .config("spark.sql.crossJoin.enabled", True)
+    .config("spark.ui.enabled", False)
+    .getOrCreate()
+)
+```
+
+## PySpark Shell
+
+```bash
+pip install pysarplus
+pyspark --packages microsoft:sarplus:0.2.6 --conf spark.sql.crossJoin.enabled=true
+```
+
+## Databricks
+
+One must set the crossJoin property to enable calculation of the similarity matrix (Clusters / &lt; Cluster &gt; / Configuration / Spark Config)
+
+```
+spark.sql.crossJoin.enabled true
+```
+
+1. Navigate to your workspace 
+2. Create library
+3. Under 'Source' select 'Maven Coordinate'
+4. Enter 'microsoft:sarplus:0.2.5' or 'microsoft:sarplus:0.2.6' if you're on Spark 2.4.1
+5. Hit 'Create Library'
+6. Attach to your cluster
+7. Create 2nd library
+8. Under 'Source' select 'Upload Python Egg or PyPI'
+9. Enter 'pysarplus'
+10. Hit 'Create Library'
+
+This will install C++, Python and Scala code on your cluster.
+
+You'll also have to mount shared storage
+
+1. Create [Azure Storage Blob](https://ms.portal.azure.com/#create/Microsoft.StorageAccount-ARM)
+2. Create storage account (e.g. <yourcontainer>)
+3. Create container (e.g. sarpluscache)
+
+1. Navigate to User / User Settings
+2. Generate new token: enter 'sarplus'
+3. Use databricks shell (installation here)
+4. databricks configure --token
+4.1. Host: e.g. https://westus.azuredatabricks.net
+5. databricks secrets create-scope --scope all --initial-manage-principal users
+6. databricks secrets put --scope all --key sarpluscache
+6.1. enter Azure Storage Blob key of Azure Storage created before
+7. Run mount code
+
+
+```pyspark
+dbutils.fs.mount(
+  source = "wasbs://sarpluscache@<accountname>.blob.core.windows.net",
+  mount_point = "/mnt/sarpluscache",
+  extra_configs = {"fs.azure.account.key.<accountname>.blob.core.windows.net":dbutils.secrets.get(scope = "all", key = "sarpluscache")})
+```
+
+Disable annoying logging
+
+```pyspark
+import logging
+logging.getLogger("py4j").setLevel(logging.ERROR)
+```
+
+
+# Packaging

```

Comment:
might be better to put this into a separate document to avoid the perception that this is needed in order to use the package

Reply:
+1

Reply:
moved out